eis_toolkit/exploratory_analyses/k_means_cluster.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+
+import geopandas as gdp
+import numpy as np
+from beartype import beartype
+from sklearn.cluster import KMeans
+from sklearn.preprocessing import StandardScaler
+
+from eis_toolkit.exceptions import EmptyDataFrameException
+
+
+def _k_means_clustering(
+    data: gdp.GeoDataFrame, number_of_clusters: int, random_state: Optional[int] = None
+) -> gdp.GeoDataFrame:
+
+    coordinates = data.geometry.apply(lambda geom: [geom.x, geom.y])
+    coordinates_array = np.array(coordinates.tolist())
+    standardized_data = StandardScaler().fit_transform(coordinates_array)
+
+    kmeans = KMeans(n_clusters=number_of_clusters, random_state=random_state)
+    kmeans.fit(standardized_data)
```

Comment:
If I'm not mistaken we don't need to standardize the coordinates in order to create a k-means model. So you could remove those rows and simply put a 2D list of coordinates to the model.
```suggestion
    coordinates = list(data.geometry.apply(lambda geom: [geom.x, geom.y]))

    kmeans = KMeans(n_clusters=number_of_clusters, random_state=random_state)
    kmeans.fit(coordinates)
```